app/api/v2/identity/users.rb

```diff
@@ -78,12 +78,18 @@ def parse_refid!
           user_params[:referral_id] = parse_refid! unless params[:refid].nil?
 
           user = User.new(user_params)
-
           code_error!(user.errors.details, 422) unless user.save
 
           activity_record(user: user.id, action: 'signup', result: 'succeed', topic: 'account')
 
-          publish_confirmation(user, Barong::App.config.domain)
+          # Creates superadmin user in first platform registration
+          if Barong::App.config.first_superadmin_registration && user.id == 1
```

Comment:
It's better you check the number of entries in database:
```
User.count == 1
```
The id would not be one if we truncate the table.